sympy/stats/random_matrix_models.py

```diff
@@ -180,13 +183,108 @@ def density(self, expr):
         return Lambda(H, exp(-S(n) * Trace(H**2))/ZGSE)
 
     def joint_eigen_distribution(self):
-        return self._compute_joint_eigen_dsitribution(4)
+        return self._compute_joint_eigen_dsitribution(S(4))
 
     def level_spacing_distribution(self):
         s = Dummy('s')
         f = ((S(2)**18)/((S(3)**6)*(pi**3)))*(s**4)*exp((-64/(9*pi))*s**2)
         return Lambda(s, f)
 
+class CircularEnsemble(RandomMatrixEnsemble):
+    """
+    Abstract class for Circular ensembles.
+    Contains the properties and methods
+    common to all the circular ensembles.
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Circular_ensemble
+    """
+    def _compute_joint_eigen_dsitribution(self, beta):
```

Comment:
dsitribution -> distribution

Reply:
There are many other spelling mistakes. Thanks for pointing out. I will correct them in the next commit.